apollo-normalized-cache/src/main/java/com/apollographql/apollo/cache/normalized/ApolloStore.kt

```diff
@@ -0,0 +1,250 @@
+   * @return {@ApolloStoreOperation} to be performed, that will be resolved with the count of records been removed
+   */
+  fun remove(cacheKeys: List<CacheKey>): ApolloStoreOperation<Int>
+
+  /**
+   * @return The [ResponseNormalizer] used to generate normalized records from the network.
+   */
+  fun networkResponseNormalizer(): ResponseNormalizer<Map<String, Any>>
+
+  /**
+   * @return The [ResponseNormalizer] used to generate normalized records from the cache.
+   */
+  fun cacheResponseNormalizer(): ResponseNormalizer<Record>
+
+  /**
+   * Run a operation inside a read-lock. Blocks until read-lock is acquired.
+   *
+   * @param transaction A code block to run once the read lock is acquired.
+   * @param <R>         The result type of this read operation.
+   * @return A result from the read operation.
+  </R> */
+  fun <R> readTransaction(transaction: Transaction<ReadableStore, R>): R
+
+  /**
+   * Run a operation inside a write-lock. Blocks until write-lock is acquired.
+   *
+   * @param transaction A code block to run once the write lock is acquired.
+   * @param <R>         The result type of this write operation.
+   * @return A result from the write operation.
+  </R> */
+  fun <R> writeTransaction(transaction: Transaction<WriteableStore, R>): R
+
+  /**
+   * @return The [NormalizedCache] which backs this ApolloStore.
+   */
+  fun normalizedCache(): NormalizedCache
+
+  /**
+   * @return the [CacheKeyResolver] used for resolving field cache keys
+   */
+  fun cacheKeyResolver(): CacheKeyResolver
+
+  /**
+   * Read GraphQL operation from store.
+   *
+   * @param operation to be read
+   * @param <D>       type of GraphQL operation data
+   * @param <T>       type operation cached data will be wrapped with
+   * @param <V>       type of operation variables
+   * @return {@ApolloStoreOperation} to be performed, that will be resolved with cached data for specified operation
+  </V></T></D> */
+  fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>): ApolloStoreOperation<T>
+
+  /**
+   * Read GraphQL operation response from store.
+   *
+   * @param operation           response of which should be read
+   * @param responseFieldMapper [ResponseFieldMapper] to be used for field mapping
+   * @param responseNormalizer  [ResponseNormalizer] to be used when reading cached response
+   * @param cacheHeaders        [CacheHeaders] to be used when reading cached response
+   * @param <D>                 type of GraphQL operation data
+   * @param <T>                 type operation cached data will be wrapped with
+   * @param <V>                 type of operation variables
+   * @return {@ApolloStoreOperation} to be performed, that will be resolved with cached response for specified operation
+  </V></T></D> */
+  fun <D : Operation.Data, T, V : Operation.Variables> read(
+      operation: Operation<D, T, V>, responseFieldMapper: ResponseFieldMapper<D>,
```

Comment:
Would it be possible to put all parameters and then the return type on a new line?

Reply:
Do you mean something like this?
```
fun <D : Operation.Data, T, V : Operation.Variables> read(
      operation: Operation<D, T, V>, responseFieldMapper: ResponseFieldMapper<D>,
      responseNormalizer: ResponseNormalizer<Record>, cacheHeaders: CacheHeaders
): ApolloStoreOperation<Response<T>>
```

Reply:
Yes. Actually every single param can be on a new line. That is what is recommended by Kotlin style guide when the function is too long.